### PR TITLE
Unit tests first

### PR DESCRIPTION
**Pull request by BertrandConxy**
# Solver-app-TDD-project: Unit tests first

Hello ✋,

In this branch, we have done the following:
- Initiated the repo
- Created solver.rb file and solver_spec.rb for tests
- Wrote tests for the required methods. Namely: find_factorial, reverse_string and fizzbuzz
- Run the tests and they failed
- Wrote codes for the class  Solver and its methods
- Run the tests again and they were passing

As we were required to do, we used TDD(Test Driven Development) approach to do this project.

### CODE REVIEW

File: Rspec_tests/solver_spec.rb [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+      expect(@solver.find_factorial(5)).to eql(120)
+    end
+
+    it 'returns 1 if argument is 0' do
+      expect(@solver.find_factorial(0)).to eql(1)
+    end
+
+    it 'raises Error when argument is negative integer' do
+      expect(@solver.find_factorial(-5)).to eql("can't work on negative integers")
+    end
```
Comment (ahmetbozaci) [inline Rspec_tests/solver_spec.rb]: - Also, you can do that for a negative value.
image

**Review by ahmetbozaci — APPROVED** [review]
# STATUS: APPROVED 🥳 🥳 🥳 



@BertrandConxy @butlermuwo  ✋🏼 
#### Your project is great. 🎉 I approve of it. There is nothing else to say other than... it's time to merge it. You can submit this project on your dashboard.

___ 

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
aproved


___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

File: Rspec_tests/solver_spec.rb (hunk 1 above)
Comment (BertrandConxy) [reply Rspec_tests/solver_spec.rb]: oh. Thank you very much. I didn't know about that marcher.

**Comment by BertrandConxy** [comment]
Thank you @ahmetbozaci

**Comment by BertrandConxy** [comment]
I have implemented it

**Pull request merged**